chainstate/src/detail/chainstateref/tx_verifier_storage.rs

```diff
@@ -269,9 +269,9 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocksRef, V: TransactionVerificati
                     .db_tx
                     .get_block_index(&id)
                     .map_err(TransactionVerifierStorageError::from)?
-                    .ok_or(
-                        TransactionVerifierStorageError::GenBlockIndexRetrievalFailed(id.into()),
-                    )?;
+                    .ok_or_else(|| {
```

Comment:
not related to your PR but just to point out the problem. I remember that I definitely run `do_checks` on this code. The difference between local runs and CI for clippy becomes annoying

Reply:
Yeah I had wild differences and that's why I made that little `python3` change so I can now run it locally :)